### PR TITLE
Realm internationalization

### PR DESCRIPTION
**Pull request by tomrutsaert**

### CODE REVIEW

File: keycloak/realm.go [hunk 1]
```diff
@@ -108,5 +113,34 @@ func (keycloakClient *KeycloakClient) ValidateRealm(realm *Realm) error {
 		return fmt.Errorf("validation error: theme \"%s\" does not exist on the server", realm.EmailTheme)
 	}
 
+	if realm.InternationalizationEnabled == false && len(realm.SupportLocales) > 0 {
+		return fmt.Errorf("validation error: SupportLocales cannot be set if InternationalizationEnabled is false")
+	}
+
+	if realm.InternationalizationEnabled == false && len(realm.DefaultLocale) > 0 {
+		return fmt.Errorf("validation error: DefaultLocale cannot be set if InternationalizationEnabled is false")
+	}
```
Comment (mrparkers) [inline keycloak/realm.go]: Are these two error messages still needed?  Based on the provider code, it looks like these simply can't be set without specifying the `internationalization {}` set.

File: keycloak/realm.go [hunk 2]
```diff
@@ -108,5 +113,34 @@ func (keycloakClient *KeycloakClient) ValidateRealm(realm *Realm) error {
...
+
+	if realm.InternationalizationEnabled == true && len(realm.SupportLocales) <= 0 {
+		return fmt.Errorf("validation error: SupportLocales should be set if Internationalization is present")
+	}
+
+	if realm.InternationalizationEnabled == true && len(realm.DefaultLocale) <= 0 {
+		return fmt.Errorf("validation error: DefaultLocale should be set if Internationalization is present")
+	}
```
Comment (mrparkers) [inline keycloak/realm.go]: These two error messages could probably be removed if these attributes were marked as `Required` in the `internationalization` set.

File: provider/resource_keycloak_realm_test.go [hunk 3]
```diff
@@ -150,6 +150,66 @@ func TestAccKeycloakRealm_themesValidation(t *testing.T) {
...
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckKeycloakRealmDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testKeycloakRealm_internationalizationValidation(realm, "en", "en"),
+				Check:  testAccCheckKeycloakRealmInternationalizationIsEnabled("keycloak_realm.realm", "en"),
+			},
```
Comment (mrparkers) [inline provider/resource_keycloak_realm_test.go]: Could you add another test step after this one that updates the internationalization settings to something different?

File: keycloak/realm.go (hunk 1 above)
Comment (tomrutsaert) [reply keycloak/realm.go]: I have removed these two.

File: keycloak/realm.go (hunk 2 above)
Comment (tomrutsaert) [reply keycloak/realm.go]: I made them required and I have removed these two messages as well.

File: provider/resource_keycloak_realm_test.go (hunk 3 above)
Comment (tomrutsaert) [reply provider/resource_keycloak_realm_test.go]: added an extra step that changes the default locale to es

**Comment by tomrutsaert** [comment]
@mrparkers I do not see ant reason why test-password ci failed or how it is related to my changes.

File: provider/resource_keycloak_realm.go [hunk 4]
```diff
@@ -229,11 +229,47 @@ func resourceKeycloakRealm() *schema.Resource {
 				Computed:         true,
 				DiffSuppressFunc: suppressDurationStringDiff,
 			},
+
+			//internationalization
+			"internationalization": {
+				Type:     schema.TypeSet,
```
Comment (mrparkers) [inline provider/resource_keycloak_realm.go]: Could you use `TypeList` like I did in #128?

Comment (tomrutsaert) [reply provider/resource_keycloak_realm.go]: done

**Pull request merged**